src/app/aoscore.hpp

```diff
@@ -81,7 +84,8 @@ class AosCore {
     common::jsonprovider::JSONProvider                                   mJSONProvider;
     common::logger::Logger                                               mLogger;
     common::oci::OCISpec                                                 mOCISpec;
-    sm::cni::CNI                                                         mCNI;
+    std::optional<sm::cni::CNI>                                          mCNI;
```

Comment:
Why is it optional?

Reply:
std::optional is needed because the CNI class has an explicit constructor that takes an executor object, and by using std::optional, the creation of the object can be deferred.

Reply:
I see. Pass executor through Init method